### PR TITLE
Add docstrings for K8sRequiredProbes

### PR DESCRIPTION
**Pull request by maxsmythe**
Signed-off-by: Max Smythe <smythe@google.com>

### CODE REVIEW

**Review by willbeason — APPROVED** [review]

File: library/general/requiredprobes/template.yaml [hunk 1]
```diff
@@ -14,10 +14,12 @@ spec:
...
           properties:
             probes:
+              description: "A list of probes that are required (ex: `readinessProbe`)"
               type: array
               items:
                 type: string
             probeTypes:
+              description: "The probe must implement one of the `probeType` in order to satisfy the constraint (ex. `tcpSocket` satisfies `['tcpSocket', 'exec']`)"
               type: array
               items:
```
Comment (julianKatz) [inline library/general/requiredprobes/template.yaml:24]: we can add an `enum` with the allowed values from [the docs](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#configure-probes)

Comment (julianKatz) [reply library/general/requiredprobes/template.yaml:24]: `readinessProbe`, `livenessProbe`

**Review by julianKatz — APPROVED** [review]

File: library/general/requiredprobes/template.yaml (hunk 1 above)
Comment (maxsmythe) [reply library/general/requiredprobes/template.yaml:24]: Unfortunately that would be a backwards-incompatible change.

Comment (maxsmythe) [reply library/general/requiredprobes/template.yaml:24]: Also, those particular values would go under `probe` as this constraint template is written.

**Review by julianKatz — APPROVED** [review]

**Review by sozercan — APPROVED** [review]

**Pull request merged**